lib/chef/resource/aws_elasticsearch_domain.rb

```diff
@@ -0,0 +1,44 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+module AWS
+  class Elasticsearch
+    class Domain
+    end
+  end
+end
+
+class Chef::Resource::AwsElasticsearchDomain < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type ::AWS::Elasticsearch::Domain
+
+  INSTANCE_TYPES = %w(m3.medium.elasticsearch m3.large.elasticsearch
```

Comment:
I'll check what the API returns.  I went with this initially because the instance types are not just EC2 instance types.  If the API returns some hint about what is se-table, I agree we should just let it fall through.